### DISCUSSION TITLE
Is there any benefit of using streamlink

### DISCUSSION THREAD

**Discussion by tathastu871** [streamlink/streamlink#5152] [General]
### Description

Players like vlc and mpv can directly play online urls 
is there any benefit to using streamlink.
because streamlink just slows down the startup.

**Comment by mkbloke** [comment]
Yes, because the Streamlink plugins:

1) Extract the playlist or manifest URL, which may not be obvious to many users (i.e. it might be retrieved via a call to a site's API)
2) Plugins implement logic to refresh authorisation tokens that may be required to enable access to streams

**Reply by tathastu871** [reply]
any benefit for direct playback.
i have direct m3u8 url or local m3u8 file.
will playing them via streamlink has any benefit over directly playing in player.

because starting streamlink to player slow down startup.
while direct playing in player is faster

**Reply by mkbloke** [reply]
Probably not, in that case.
